### PR TITLE
Add Symfony Messenger Only

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | Y
| Deprecations?                          | N
| BC breaks? (use the c.x branch)        | N
| Automated tests included?              | Y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes  

#### Description:

I think PR #11180 is too big and it makes it hard for other developers to test and review file changes, I made this new PR to install `symfony/messenger` as a new bundle and install the required packages along with the configuration UI for the new bundle. **Once we get this PR approved and merged, I will make a new PR that will contain messenger alone**

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to the configuration page, check for `Messenger Configuration`
3. Go to `local.php` and see the values of `messenger_dsn` changes according to the settings you fill
4. If you disabled Messenger you should see in `local.php` `messenger_type`  => `sync`
5. Messenger configuration for doctrine variables can be found on https://symfony.com/doc/current/messenger.html#doctrine-transport

### CODE REVIEW

File: app/config/config.php [hunk 1]
```diff
@@ -83,6 +83,30 @@
...
+    'messenger'            => [
+        'default_bus' => 'email.bus',
+        'buses'       => [
+            'email.bus' => null,
+        ],
+        'transports'  => [
+            'email_transport' => [
+                'dsn'            => '%env(MAUTIC_MESSENGER_TRANSPORT_DSN)%',
+                'options'        => [
+                    'table_name' => MAUTIC_TABLE_PREFIX.'messenger_messages',
```
Comment (mabumusa1) [inline app/config/config.php:96]: I am using this custom definition of the table name to include the prefix of the table according to these [docs](https://symfony.com/doc/4.4/messenger.html#doctrine-transport)

**Comment by RCheesley** [comment]
Tried to test this but Mautic is not installed:

```
Installing Mautic...
Mautic Install
==============

Parsing options and arguments...
0 - Checking installation requirements...
Missing optional settings:
  - [0] Configure <code>zend.assertions = -1</code> in php.ini for production environments for faster and more stable execution.
Continue with install anyway? [yes/no]Ready to Install!
1 - Creating database...
1.1 - Creating schema...
1.2 - Loading fixtures...
2 - Creating admin user...
3 - Email configuration...
PHP Warning:  Undefined array key "messenger_transport" in /var/www/html/app/bundles/MessengerBundle/Helper/MessengerDsnConvertor.php on line 40
PHP Warning:  Undefined array key "messenger_transport" in /var/www/html/app/bundles/MessengerBundle/Helper/MessengerDsnConvertor.php on line 50
PHP Warning:  Undefined array key "" in /var/www/html/app/bundles/MessengerBundle/Helper/MessengerDsnConvertor.php on line 50

In MessengerDsnConvertor.php line 51:
                           
  Class "" does not exist  
```

**Comment by mabumusa1** [comment]
> messenger_transport

Thanks, @RCheesley for reporting, I just fixed it let's wait until the checks are green then you can test again

**Comment by RCheesley** [comment]
Still not installing correctly:

```
3 - Email configuration...
PHP Warning:  Undefined array key "messenger_transport" in /var/www/html/app/bundles/MessengerBundle/Helper/MessengerDsnConvertor.php on line 40
PHP Warning:  Undefined array key "messenger_transport" in /var/www/html/app/bundles/MessengerBundle/Helper/MessengerDsnConvertor.php on line 50
PHP Warning:  Undefined array key "" in /var/www/html/app/bundles/MessengerBundle/Helper/MessengerDsnConvertor.php on line 50

In MessengerDsnConvertor.php line 51:
                           
  Class "" does not exist  
                           

mautic:install [-f|--force] [--db_driver DB_DRIVER] [--db_host DB_HOST] [--db_port DB_PORT] [--db_name DB_NAME] [--db_user DB_USER] [--db_password DB_PASSWORD] [--db_table_prefix DB_TABLE_PREFIX] [--db_backup_tables DB_BACKUP_TABLES] [--db_backup_prefix DB_BACKUP_PREFIX] [--admin_firstname ADMIN_FIRSTNAME] [--admin_lastname ADMIN_LASTNAME] [--admin_username ADMIN_USERNAME] [--admin_email ADMIN_EMAIL] [--admin_password ADMIN_PASSWORD] [--mailer_from_name [MAILER_FROM_NAME]] [--mailer_from_email [MAILER_FROM_EMAIL]] [--mailer_transport [MAILER_TRANSPORT]] [--mailer_host MAILER_HOST] [--mailer_port MAILER_PORT] [--mailer_user MAILER_USER] [--mailer_password [MAILER_PASSWORD]] [--mailer_encryption [MAILER_ENCRYPTION]] [--mailer_auth_mode [MAILER_AUTH_MODE]] [--mailer_spool_type MAILER_SPOOL_TYPE] [--mailer_spool_path MAILER_SPOOL_PATH] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command> <site_url> [<step>]
```

Using Gitpod: https://gitpod.io/#https://github.com/mautic/mautic/pull/11598 - it doesn't complete the installation.

**Comment by RCheesley** [comment]
Noticed this in the logs while it was installing Mautic:

```
Errors in email configuration:
  - [messneger_transport] A value is required
```

Maybe an issue with a typo?

**Comment by volha-pivavarchyk** [comment]
My installation in gitpod ended with an error:
image

And one more small thing. There is no translation
image

**Comment by mabumusa1** [comment]
@RCheesley @volha-pivavarchyk Thank you for testing this PR, it should be good to go in both CLI and web installation

**Review by RCheesley — APPROVED** [review]
Install ✅ 
Configure in settings ✅ 

Currently I was not able to get it to use the queue but discussing with @mabumusa1 this PR is just to introduce Messenger, and doesn't actually wire up the queueing system just yet - that will come in a separate PR as otherwise it introduces too much all at once.

So all the things which weren't working before are now fixed and I was able to install and configure.

Only question: I think this should be higher in the settings list - maybe around the Email section - rather than the last thing in the list.

@escopecz do we have any precedents for how we order things in the settings area?

**Comment by escopecz** [comment]
@RCheesley what do you mean by "this" in

> this should be higher in the settings lis

There is no UI/UX overview other than asking for changes in PR comments as you are doing. There is no rule for it.

**Comment by RCheesley** [comment]
@escopecz when you check the configuration, Messenger is right at the very bottom below all the other bundles in the user interface.

screenshot-8080-mabumusa1-mautic-n9np4lbvnju ws-eu74 gitpod io-2022 11 18-15_16_03

As this relates to email/messaging, and you could also say it's a system setting, I was asking whether it should be higher in the tabs on the left, near to the settings where you configure these already, instead of dumped at the bottom to disappear into obscurity. Or whether it should be within system settings itself, I guess, might be another question.

As a user, I'd expect to see all the things related to channels in the same area, rather than having some at the top, some at the bottom. Right now, it's not a great experience. Appreciate it might be a wider question to address outside this PR and something we can fix outside this PR (if so we should raise an issue) but as we are adding another section, that is what triggered my question as to what our decision making process is for adding and positioning new sections.

To summarise, my question is, should we think about how we organise the settings in the user interface now we're adding a new one? Or do we just dump new things at the bottom of the list without thinking how the user experience would be for the person configuring?

**Review by volha-pivavarchyk — APPROVED** [review]
The settings are changed properly as discribed.
I have tested just steps. It is only the settings.

**Comment by RCheesley** [comment]
My comments on the UX should not be seen to be merge-blocking - it is something we can address with the UI/UX tiger team at a later date. Marking it as RTC.

**Review by escopecz — APPROVED** [review]
Thanks! This is good first step.
